### ISSUE TITLE
error after consuming all gas

### ISSUE THREAD

**Issue by junaidev** [trufflesuite/truffle#2342]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue
Compiling your contracts...
> Everything is up to date, there is nothing to compile.

Starting migrations...
> Network name:    'development'
> Network id:      273
> Block gas limit: 0x7a1200

1_initial_migration.js
   Deploying 'Migrations'
   > transaction hash:    0x5a17c8e3e2884c37517c784171246ef5f88424ad91a466e0b38fa9c113d5467f
   > Blocks: 0            Seconds: 0
   > contract address:    0xb6ebe346101eA657a0046d7378Aa6612A1FA8920
   > block number:        36
   > block timestamp:     1566994125
   > account:             0x4A57Bbf8666E532eDA673f8d76C90e13F1bc9371
   > balance:             1606938044258990275541962092341162602522202.521509812835301376
   > gas used:            261329
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00522658 ETH

   ⠋ Saving migration to chain.Error: Error: StatusError: Transaction: 0x929867ffdc2244cdc08e4e4f850c385774eead3f80815d661d5aee714dd94192 exited with an error (status 0) after consuming all gas.
     Please check that the transaction:
     - satisfies all conditions set by Solidity `assert` statements.
     - has enough gas to execute the full transaction.
     - does not trigger an invalid opcode by other means (ex: accessing an array out of bounds).
    at Object.run (/home/usra/.npm-packages/lib/node_modules/truffle/build/webpack:/packages/truffle-migrate/index.js:92:1)
    at process._tickCallback (internal/process/next_tick.js:68:7)

## Steps to Reproduce

I am able to reproduce on my system using:
npm install truffle -g
truffle init
npm init
then added:
` development: {
      host: "127.0.0.1",     // Localhost (default: none)
      port: 8545,            // Standard Ethereum port (default: none)
      network_id: "*",       // Any network (default: none)
      from: "0x4a57bbf8666e532eda673f8d76c90e13f1bc9371"
     }`
this account in unlocked on instant seal local parity. 

## Expected Behavior

it should deploy at least migration contract

## Actual Results
its failing with error described above.

## Environment

* Operating System: Linux Mint 19
* Ethereum client: Parity v2.6.1-beta
*Truffle v5.0.33 (core: 5.0.33)
*Solidity v0.5.0 (solc-js)
*Node v10.16.3
*Web3.js v1.2.1
*npm: 6.9.0

**Comment by junaidev** [comment]
down graded truffle using:
npm i -g truffle@v5.0.5 its working, but its failing with latest truffle

**Comment by junaidev** [comment]
parity:
https://github.com/paritytech/parity-ethereum/releases/tag/v2.6.1 
let me test it with ganache as well.

**Comment by eggplantzzz** [comment]
Yes I know parity but I was mainly asking about your setup so that I can run it locally using the same configuration as you.

**Comment by hunterperm** [comment]
Was this ever resolved? We are having the same issue I believe. Works with v5.0.5 but gas calculation error vith v5.0.6+

**Comment by haltman-at** [comment]
@hunterperm, could you provide us with more information?  Do you have reproduction steps?  (To confirm, this is still happening on the most recent Truffle, yes?)  What leads you to say it's a gas calculation error -- can you be more detailed about just what's different between 5.0.5 and 5.0.6?  Thank you!

**Comment by fainashalts** [comment]
@junaidev we are closing this for issue maintenance. Please open a new issue if this is still something that needs our attention. Thanks!

**Comment by danielspk** [comment]
The same is still happening in version 5.1.48 when using @truffle/hdwallet-provider
